File: app/src/main/java/com/example/android/learnarchitecturecomponents/livedata/NameViewModel.java
package com.example.android.learnarchitecturecomponents.livedata;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

/**
 * Created by dmw on 2018/9/15.
 * Desc:
 */
public class NameViewModel extends ViewModel {

    //创建一个持有String类型数据的LiveData
    private MutableLiveData<String> mCurrentName;

    public MutableLiveData<String> getCurrentName() {
        if (mCurrentName == null) {
            mCurrentName = new MutableLiveData<>();
        }
        return mCurrentName;
    }
}

File: app/src/main/java/com/example/architecturecomponents/livedata/NameViewModel.kt
package com.example.architecturecomponents.livedata

import androidx.lifecycle.MutableLiveData

import androidx.lifecycle.ViewModel

/**
 * Created by dmw on 2018/9/15.
 * Desc:
 */
class NameViewModel : ViewModel() {

    //创建一个持有String类型数据的LiveData
    val currentName: MutableLiveData<String> by lazy {
        MutableLiveData<String>()
    }

}
